test/e2e/util_gatewayapi_test.go

```diff
@@ -272,6 +324,71 @@ func buildHTTPRoute(routeName, namespace, parentgateway, parentNamespace, hostna
 	}
 }
 
+// buildGWAPICRDFromName initializes the GatewayAPI CRD deducing most of its required fields from the given name.
+func buildGWAPICRDFromName(name string) *apiextensionsv1.CustomResourceDefinition {
+	var (
+		plural   = strings.Split(name, ".")[0]
+		group, _ = strings.CutPrefix(name, plural+".")
+		scope    = apiextensionsv1.NamespaceScoped
+		// removing trailing "s"
+		singular = plural[0 : len(plural)-1]
+		kind     string
+	)
+
+	switch plural {
+	case "gatewayclasses":
+		singular = "gatewayclass"
+		kind = "GatewayClass"
+		scope = apiextensionsv1.ClusterScoped
+	case "gateways":
+		kind = "Gateway"
+	case "httproutes":
+		kind = "HTTPRoute"
+	case "referencegrants":
+		kind = "ReferenceGrant"
+	}
```

Comment:
Also `GRPCRoute`.  Is it in our best interest to implement this function, knowing it has to change in the future every time a new CRD is added or the version changed?

Reply:
> Is it in our best interest to implement this function, knowing it has to change in the future every time a new CRD is added or the version changed?

Well, yes, we will have to come to this function for every new type just like `crdNames` we discussed in [this comment](https://github.com/openshift/cluster-ingress-operator/pull/1192#discussion_r1984117536). And I think it's the good thing for the test because the test should not rely on the packages used in the code to avoid duplication of erroneous logic (same reason as in [my response](https://github.com/openshift/cluster-ingress-operator/pull/1192#discussion_r1984753210) for the before mentioned comment).